From 417f31f7af3c6e9c03bf62d29e6e0b9285e3dc6c Mon Sep 17 00:00:00 2001
From: yann_dm <yann.montmarin@gmail.com>
Date: Fri, 7 Oct 2022 10:05:19 +0200
Subject: [PATCH 2/3] fix: boost include

---
 src/serialization.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git src/serialization.cc src/serialization.cc
index 0a1a09f..bdd4a91 100644
--- src/serialization.cc
+++ src/serialization.cc
@@ -29,6 +29,9 @@
 
 #include <boost/serialization/list.hpp>
 #include <boost/serialization/serialization.hpp>
+#if __has_include(<boost/serialization/library_version_type.hpp>)
+  #include <boost/serialization/library_version_type.hpp>
+#endif
 #include <boost/serialization/set.hpp>
 #include <boost/serialization/weak_ptr.hpp>
 #include <hpp/core/connected-component.hh>
-- 
2.25.1

