From ff999768234747d6b50fe7bdcf750ed9bc7214d9 Mon Sep 17 00:00:00 2001
From: yann_dm <yann.montmarin@gmail.com>
Date: Fri, 7 Oct 2022 10:05:54 +0200
Subject: [PATCH 3/3] fix: explicit rt linking

---
 tests/CMakeLists.txt | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git tests/CMakeLists.txt tests/CMakeLists.txt
index ef02326..2b8881d 100644
--- tests/CMakeLists.txt
+++ tests/CMakeLists.txt
@@ -46,6 +46,9 @@ ADD_DEFINITIONS(-DTEST_DIRECTORY="${CMAKE_SOURCE_DIR}/tests/data")
 # This macro will create a binary from `NAME.cc', link it against
 # Boost and add it to the test suite.
 #
+
+find_library(RT_LIB rt)
+
 MACRO(ADD_TESTCASE NAME GENERATED)
   IF (${GENERATED} STREQUAL TRUE)
     ADD_UNIT_TEST(${NAME} ${CMAKE_CURRENT_BINARY_DIR}/${NAME}.cc)
@@ -53,7 +56,11 @@ MACRO(ADD_TESTCASE NAME GENERATED)
     ADD_UNIT_TEST(${NAME} ${NAME}.cc)
   ENDIF()
   TARGET_INCLUDE_DIRECTORIES(${NAME} PRIVATE ../src Boost::unit_test_framework)
-  TARGET_LINK_LIBRARIES(${NAME} ${PROJECT_NAME}-gpl)
+  if(RT_LIB)
+    TARGET_LINK_LIBRARIES(${NAME} ${PROJECT_NAME}-gpl ${RT_LIB})
+  else()
+    TARGET_LINK_LIBRARIES(${NAME} ${PROJECT_NAME}-gpl)
+  endif()
 ENDMACRO(ADD_TESTCASE)
 
 FIND_PACKAGE(OpenMP)
-- 
2.25.1

